index.js

```diff
@@ -0,0 +1,161 @@
+'use strict'
+
+const decode = Symbol('decode')
+const encode = Symbol('encode')
+
+function parsePair (segment) {
+  return segment.trim().split('=')
+}
+
+class SetCookie {
+  constructor (input, options) {
+    if (Array.isArray(input)) {
+      return input.map(item => new SetCookie(item, options))
+    }
+
+    options = options || {}
+    let data = null
+
+    // Options
+    this[decode] = options.decode || decodeURIComponent
+    this[encode] = options.encode || encodeURIComponent
+
+    // Convert strings to objects
+    if (typeof input === 'string') {
+      const segments = input.split(';').map(parsePair)
+      const pair = segments.shift()
+
+      if (pair.length < 2 || !pair[0] || !pair[1]) {
+        throw new Error('Invalid key-value pair')
+      }
+
+      data = {
+        key: this[decode](pair[0]),
```

Comment:
From https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Set-Cookie:

- `__Secure-` prefix: Cookies with a name starting with `__Secure-` (dash is part of the prefix) must be set with the `secure` flag and must be from a secure page (HTTPS).
- `__Host-` prefix: Cookies with a name starting with `__Host-` must be set with the `secure` flag, must be from a secure page (HTTPS), must not have a domain specified (and therefore aren't sent to subdomains) and the path must be "/".

Reply:
Did you consider this?